#!/usr/bin/env python
PACKAGE = "franka_interactive_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 0, 0, 600)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 30, 0, 60)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.01, 0, 100)
gen.add("activate_tool_compensation", bool_t, 0, "trigger to activate external tool compensation (the desired compensation forces defined in ./config)", True)
gen.add("update_impedance_params", bool_t, 0, "Will update the impedance parameters when set to True", False)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
